view/frontend/templates/analytics/initialize.phtml

```diff
@@ -1,24 +1,14 @@
 <?php if ( $this->isKlaviyoEnabled() && $this->getPublicApiKey() ): ?>
+<script type="application/javascript" async 
+ src="https://static.klaviyo.com/onsite/js/klaviyo.js?company_id=<?php echo $this->getPublicApiKey(); ?>"></script>
```

Comment:
It looks like `getPublicApiKey()` (and potentially other methods) perform DB requests. It means every page load introduces an extra load. As far as I remember, Magento has cache and it should be possible to cache these values.